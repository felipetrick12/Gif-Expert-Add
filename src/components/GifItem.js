import React from 'react'
import PropTypes from "prop-types";


export default function GifItem({title,url}) {

    return (
        
        <div className='card animate__animated animate__zoomInDown' >
           <p>{title}</p>
           <img src={url} alt={title}/>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  };
