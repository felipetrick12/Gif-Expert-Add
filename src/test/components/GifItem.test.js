import { shallow } from "enzyme";
import React from 'react';
import GifItem from "../../components/GifItem";


describe('Prueba de Componente GifItem', () => {
    
    const url = 'https://www.facebook.com';
    const title='soy rick';
    const clas='animate__zoomInDown';
    const wapper = shallow(<GifItem title={title} url={url}/>);

    test('Deberia renderizar el componente ', () => {

        expect(wapper).toMatchSnapshot();    

    });

    test('deberia tener un parrafo con titulo ', () => {

            const p = wapper.find('p'); //busque en el componenente gifitem,(wapper), el p del componente
            expect(p.text().trim()).toBe(title);//si hay un p,en texto quitele los espacios y comparelo con el titulo que e envie si son iguales
    });

    test('deberia tener una url y un alt en la img', () => {
        
        const img = wapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Deberia habr un Div con la clase animate__zoomInDown', () => {
        
        const div= wapper.find('div');

        expect(div.prop('className').includes(clas)).toBe(true);
       
    });
    
    
    
    
})
