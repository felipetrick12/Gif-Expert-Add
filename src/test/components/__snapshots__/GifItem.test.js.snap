// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de Componente GifItem Deberia renderizar el componente  1`] = `
<div
  className="card animate__animated animate__zoomInDown"
>
  <p>
    soy rick
  </p>
  <img
    alt="soy rick"
    src="https://www.facebook.com"
  />
</div>
`;
